### PR TITLE
chore(mem_allocator): JEAllocator -> AllocatorProxy

### PR DESCRIPTION
**Pull request by BohuTANG**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

* better the name from `JEAllocator -> AllocatorProxy`

Closes #issue

### CODE REVIEW

**Comment by BohuTANG** [comment]
`mem_allocator` crate is not clear enough and needs some refactoring work.

File: src/common/base/src/mem_allocator/je_allocator.rs [hunk 1]
```diff
@@ -80,7 +64,7 @@ pub mod linux_or_macos {
         }
     }
 
-    unsafe impl<T: Allocator> Allocator for JEAllocator<T> {
+    unsafe impl<T: Allocator> Allocator for AllocatorProxy<T> {
```
Comment (zhang2014) [inline src/common/base/src/mem_allocator/je_allocator.rs:67]: implementation is jemalloc? maybe JEAllocator is better?

Comment (BohuTANG) [reply src/common/base/src/mem_allocator/je_allocator.rs:67]: I think we have two layer wrappers is bit confused here: 
`JEAllocator ` is a general wrapper(not a `jemalloc` wrapper), it is defined as:
```
#[derive(Debug, Clone, Copy, Default)]
pub struct JEAllocator<T> {
    #[allow(dead_code)]
    allocator: T,
}

impl<T> JEAllocator<T> {
    pub fn new(allocator: T) -> Self {
        Self { allocator }
    }
}
```

`MmapAllocator` also define his wrapper(It's a general too):
```
pub struct MmapAllocator<T> {
    allocator: T,
}

impl<T> MmapAllocator<T> {
    pub fn new(allocator: T) -> Self {
        Self { allocator }
    }
}
```

At final, we defined `GlobalAllocator` as follows:
 ```
pub struct GlobalAllocator

type Fallback = JEAllocator<SystemAllocator>;
````
and using the `JEAllocator`:
```
#[global_allocator]
pub static GLOBAL_ALLOCATOR: GlobalAllocator = GlobalAllocator;
```

How can we change the allocator from `jemalloc` to `mmap` or a new custom allocator? 

The better way is we define `GlobalAllocator` as only allocator wrapper:
```
#[derive(Debug, Clone, Copy, Default)]
pub struct GlobalAllocator<T> {
    #[allow(dead_code)]
    allocator: T,
}

impl<T> GlobalAllocator<T> {
    pub fn new(allocator: T) -> Self {
        Self { allocator }
    }
}
```
And we use it as:
```
#[global_allocator]
pub static GLOBAL_ALLOCATOR: GlobalAllocator<JeAllocator> = GlobalAllocator<JeAllocator>;
```

or 
```
#[global_allocator]
pub static GLOBAL_ALLOCATOR: GlobalAllocator<MmapAllocator> = GlobalAllocator<MmapAllocator>;
```

If missing something, please correct me :)

**Comment by BohuTANG** [comment]
Closed, still need to research deeply.

**Pull request closed without merging**